pkg/concepts/type.go

```diff
@@ -157,7 +165,7 @@ func (t *Type) IsMap() bool {
 
 // IsScalar returns true iff this type is an scalar type.
 func (t *Type) IsScalar() bool {
-	return t.kind == ScalarType || t.kind == EnumType
+	return t.kind == ScalarType || t.kind == EnumType || t.kind == InterfaceType
```

Comment:
Maybe it's worth explaining in the doc comment what you mean by "scalar".  An interface will usually contain a slice or map, which are not conventionally understood as scalars.

Is the idea that an interface is opaque as far as the  SDK is concerned?

Reply:
> Is the idea that an interface is opaque as far as the SDK is concerned?

I guess from the SDK point of view, handling `interface{}` types requires no knowledge of its structure. Somewhat similar to `time.Time`, which is arguably not scalar in Go, yet we bundle it in the `ScalarType`, and the SDK doesn't care about it since it just treats it as any other native type from the point of view of code generation.

Maybe you're onto something here.

To be honest, I (re)used the scalar metatype because it has the highest overlap in code generation. Defining a new complex interface type is a _ton_ of hard-to-test code, and then using it means a lot of code duplication, most of which ends up overlapping with the ScalarType anyway... so the majority of our templates will end up with `{{ if or .Type.IsScalar .Type.IsInterface }}`

We could rename the "scalar" to something else to make it more explicit, but not entirely sure what that should be...

Now, this is assuming all this code works well (see openshift-online/ocm-api-model/pull/200). If it turns out that the generated SDK has a bunch of hidden bugs, then this all goes out the window and we will need a complex `InterfaceType` with its associated code flow.

Reply:
If it works, I'm OK with it :+1:. Just explain something about it in the doc comment.

A better name could be nice but naming is hard...  `IsOpaque`?  kinda more accurate but not sure it's clearer.

Reply:
`IsOpaque` is a bit more... opaque? :laughing: 
`IsBuiltIn` might also work (based on @jhernand's comment: `// Add the built-in scalar types:` when adding all the scalar types).

For now I've just added extra comments explaining why interface is treated as scalar. I'm happy to revisit next time someone starts asking questions :no_bicycles: